src/Translation_unit.ml

```diff
@@ -230,6 +230,17 @@ let check_all_locations fmt cmts_t =
           "Warning: Some locations have not been considered\n%!" ;
         List.iter ~f:print (List.sort l ~compare:Location.compare)
 
+let with_optional_box_debug ~box_debug k fs =
+  if box_debug then Fmt.with_box_debug k fs else k fs
```

Comment:
You shouldn't take `fs` argument here, `with_box_debug` is `t -> t`.

Reply:
good idea!